realm/src/main/java/io/realm/Realm.java

```diff
@@ -1206,7 +1206,12 @@ public void addChangeListener(RealmChangeListener listener) {
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
-        changeListeners.remove(new WeakReference<RealmChangeListener>(listener));
+        Iterator<WeakReference<RealmChangeListener>> iterator = changeListeners.iterator();
+        while (iterator.hasNext()) {
```

Comment:
Another thing, do you think check duplicated entries in addChangeListener would be a good idea? It would prevent some strange problem caused by user's mistake, and we can stop the loop immediately in removeChangeListener.